tools/cli/commands/build.js

```diff
@@ -768,13 +768,39 @@ async function buildProject( t ) {
 		if ( composerJson.repositories.length === 0 ) {
 			delete composerJson.repositories;
 		}
-		await writeFileAtomic(
-			`${ buildDir }/composer.json`,
-			JSON.stringify( composerJson, null, '\t' ) + '\n',
-			{ encoding: 'utf8' }
-		);
 	}
 
+	// Update dependency version numbers in composer.json.
+	const composerDepTyes = [ 'require', 'require-dev' ];
+	for ( const key of composerDepTyes ) {
+		if ( composerJson[ key ] ) {
+			for ( const [ pkg ] of Object.entries( composerJson[ key ] ) ) {
+				Object.values( t.ctx.versions ).find( ctxPkg => {
```

Comment:
This seems like a weird way to use `.find()`. If you're going to use it, I'd expect the code to look something like this:

```js
const ctxPkg = Object.values( t.ctx.versions ).find( p => p.name === pkg );
if ( ctxPkg ) {
    ...
}
```
When you're doing the actual work inside there, it would make more sense to use `.forEach()` instead. Or just make it a `for ( const ctxPkg of Object.values( t.ctx.versions ) ) {}` loop.